File: src/main/java/com/dansoftware/libraryapp/gui/login/quick/QuickLoginView.java
package com.dansoftware.libraryapp.gui.login.quick;

import com.dansoftware.libraryapp.db.DatabaseMeta;
import com.dansoftware.libraryapp.gui.context.Context;
import com.dansoftware.libraryapp.gui.context.ContextTransformable;
import com.dansoftware.libraryapp.gui.login.DatabaseLoginListener;
import com.dansoftware.libraryapp.gui.util.ImprovedFXMLLoader;
import com.dansoftware.libraryapp.i18n.I18N;
import com.dlsc.workbenchfx.SimpleHeaderView;
import de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIcon;
import de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIconView;
import javafx.scene.layout.Region;
import javafx.scene.layout.VBox;
import org.jetbrains.annotations.NotNull;

public class QuickLoginView extends SimpleHeaderView<Region> implements ContextTransformable {

    private final Context asContext;

    public QuickLoginView(@NotNull DatabaseMeta databaseMeta, @NotNull DatabaseLoginListener loginListener) {
        super(buildTitle(databaseMeta), new MaterialDesignIconView(MaterialDesignIcon.LOGIN));
        this.asContext = Context.from(this);
        this.setContent(loadContent(databaseMeta, loginListener));
    }

    private VBox loadContent(@NotNull DatabaseMeta databaseMeta, @NotNull DatabaseLoginListener loginListener) {
        return new ImprovedFXMLLoader(
                new QuickFormController(asContext, databaseMeta, loginListener),
                getClass().getResource("Form.fxml"), I18N.getValues()).load();
    }

    private static String buildTitle(@NotNull DatabaseMeta databaseMeta) {
        return String.format("%s - %s", I18N.getValue("login.quick.title"), databaseMeta);
    }

    @Override
    public @NotNull Context getContext() {
        return asContext;
    }
}

File: src/main/java/com/dansoftware/boomega/gui/login/quick/QuickLoginView.kt
/*
 * Boomega
 * Copyright (C)  2021  Daniel Gyoerffy
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package com.dansoftware.boomega.gui.login.quick

import com.dansoftware.boomega.db.DatabaseMeta
import com.dansoftware.boomega.gui.context.Context
import com.dansoftware.boomega.gui.context.ContextTransformable
import com.dansoftware.boomega.gui.login.DatabaseLoginListener
import com.dansoftware.boomega.i18n.I18N
import com.dlsc.workbenchfx.SimpleHeaderView
import de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIcon
import de.jensd.fx.glyphs.materialdesignicons.MaterialDesignIconView
import javafx.scene.layout.Region

class QuickLoginView(databaseMeta: DatabaseMeta, loginListener: DatabaseLoginListener) : SimpleHeaderView<Region?>(
    "${I18N.getValue("login.quick.title")} - $databaseMeta", MaterialDesignIconView(MaterialDesignIcon.LOGIN)
), ContextTransformable {
    private val asContext: Context = Context.from(this)

    init {
        content = QuickForm(asContext, databaseMeta, loginListener)
    }

    override fun getContext(): Context {
        return asContext
    }
}